### PR TITLE
iox-#1196 Introduce and fix warnings in posh tests

### PR DESCRIPTION
**Pull request by FerdinandSpitzschnueffler**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer
This PR introduces the warnings in CMakeLists.txt for posh tests and fixes them.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Partially closes #1196

### CODE REVIEW

File: iceoryx_posh/test/integrationtests/test_client_server.cpp [hunk 1]
```diff
@@ -271,7 +271,8 @@ TEST_F(ClientServer_test, SlowServerDoesNotBlockClient)
     Server<DummyRequest, DummyResponse> server{sd, serverOptions};
 
     // send request till queue overflows
-    for (uint64_t i = 0; i < serverOptions.requestQueueCapacity + NUMBER_OF_OVERFLOWS; ++i)
+    const int64_t iMax = static_cast<int64_t>(serverOptions.requestQueueCapacity) + NUMBER_OF_OVERFLOWS;
```
Comment (MatthiasKillat) [inline iceoryx_posh/test/integrationtests/test_client_server.cpp:274]: I think we could avoid the problem entirely if `NUMBER_OF_OVERFLOWS` would be `uint64_t` instead?

Logically the tests should be still ok with this change.

File: iceoryx_posh/test/mocks/trigger_handle_mock.hpp [hunk 2]
```diff
@@ -47,20 +47,34 @@ class MockTriggeHandle
         return *this;
     }
 
+#ifdef __clang__
```
Comment (MatthiasKillat) [inline iceoryx_posh/test/mocks/trigger_handle_mock.hpp:50]: The more I see this `#pragma` sequence the more I think we should define some abbreviation macro for this (or let the compiler ignore it on a global level if possible. Can be decided in a separate PR though, fine for now.

File: iceoryx_posh/test/moduletests/test_popo_publisher.cpp [hunk 3]
```diff
@@ -214,7 +214,7 @@ TEST_F(PublisherTest, CanLoanSamplesAndPublishTheResultOfFunctionPointerWithAddi
         .WillOnce(Return(ByMove(iox::cxx::success<iox::mepoo::ChunkHeader*>(chunkMock.chunkHeader()))));
     EXPECT_CALL(portMock, sendChunk(chunkMock.chunkHeader()));
     // ===== Test ===== //
-    auto result = sut.publishResultOf(freeFunctionWithAdditionalArgs, 42, 77.77);
+    auto result = sut.publishResultOf(freeFunctionWithAdditionalArgs, 42U, 77.77f);
```
Comment (MatthiasKillat) [inline iceoryx_posh/test/moduletests/test_popo_publisher.cpp:217]: `77.77F` for consistency (but I do not like the upper case style).

File: iceoryx_posh/test/moduletests/test_roudi_process.cpp [hunk 4]
```diff
@@ -47,7 +47,7 @@ class Process_test : public Test
 {
   public:
     const iox::RuntimeName_t processname = {"TestProcess"};
-    pid_t pid{200U};
+    uint32_t pid{200U};
```
Comment (MatthiasKillat) [inline iceoryx_posh/test/moduletests/test_roudi_process.cpp:50]: Why is this change required? `pid_t` is probably from some of the included headers and signed. (Did not check).

The question is: is it important for subsequent tests that the pid is of type `pid_t`? I do not know. But since the tests do not break for now it is probably ok.

File: iceoryx_posh/test/moduletests/test_roudi_process_manager.cpp [hunk 5]
```diff
@@ -55,7 +55,7 @@ class ProcessManager_test : public Test
     }
 
     const iox::RuntimeName_t m_processname{"TestProcess"};
-    const pid_t m_pid{42U};
+    const uint32_t m_pid{42U};
```
Comment (MatthiasKillat) [inline iceoryx_posh/test/moduletests/test_roudi_process_manager.cpp:58]: Same question as before (why required and do we understand the implications?).

**Review by MatthiasKillat — COMMENTED** [review]
OK, I am just not sure about the `pid_t` change.

File: iceoryx_posh/test/moduletests/test_roudi_process_manager.cpp (hunk 5 above)
Comment (FerdinandSpitzschnueffler) [reply iceoryx_posh/test/moduletests/test_roudi_process_manager.cpp:58]: `pid` is in this file only needed for the `ProcessManager::registerProcess` call which requires an `uint32_t`.

File: iceoryx_posh/test/moduletests/test_fixed_size_container.cpp [hunk 6]
```diff
@@ -115,27 +115,29 @@ TEST_F(FixedSizeContainer_test, removeAndSizeMultiElementContainer)
...
         size_t newSize = 2 * (k + 1);
         if (newSize > 100)
+        {
             newSize = 100;
+        }
 
         EXPECT_THAT(container.size(), Eq(newSize));
     }
 
     for (uint32_t k = 0; k < capacity; ++k)
```
Comment (elfenpiff) [inline iceoryx_posh/test/moduletests/test_fixed_size_container.cpp:138]: U

**Review by elfenpiff — DISMISSED** [review]

File: iceoryx_posh/test/moduletests/test_mq_message.cpp [hunk 7]
```diff
@@ -78,7 +78,7 @@ TEST_F(IpcMessage_test, CTorWithInitializerList_validEntries)
     IpcMessage message3({"", "", ""});
     EXPECT_THAT(message3.isValid(), Eq(true));
     EXPECT_THAT(message3.getNumberOfElements(), Eq(3u));
-    for (int i = 0; i < 3; ++i)
+    for (uint16_t i = 0; i < 3; ++i)
```
[the remark this answers is not in the document]
Comment (FerdinandSpitzschnueffler) [reply iceoryx_posh/test/moduletests/test_mq_message.cpp:81]: I prefer to leave it as it is, see https://github.com/eclipse-iceoryx/iceoryx/discussions/617

[the remark this answers is not in the document]
Comment (MatthiasKillat) [reply iceoryx_posh/test/moduletests/test_mq_message.cpp:81]: Yes. I prefer this as well.

**Review by MatthiasKillat — APPROVED** [review]

File: iceoryx_posh/test/integrationtests/test_client_server.cpp (hunk 1 above)
Comment (MatthiasKillat) [reply iceoryx_posh/test/integrationtests/test_client_server.cpp:274]: ... oh well

**Review by MatthiasKillat — APPROVED** [review]

**Review by elfenpiff — APPROVED** [review]

**Review by MatthiasKillat — APPROVED** [review]
